File: MTLab4ChangingOrder/Table.cs
namespace MTLab4ChangingOrder;

public class Table
{
    private readonly SemaphoreSlim[] _forks = new SemaphoreSlim[5];

    public Table()
    {
        for (var i = 0; i < _forks.Length; i++)
        {
            _forks[i] = new SemaphoreSlim(1);
        }
    }

    public void TakeFork(int forkIndex)
    {
        _forks[forkIndex].Wait();
    }

    public void ReleaseFork(int forkIndex)
    {
        _forks[forkIndex].Release();
    }
}

File: MTLab4JavaChangingOrder/src/Table.java
import java.util.concurrent.Semaphore;

public class Table {
    private final Semaphore[] forks = new Semaphore[5];

    public Table() {
        for (int i = 0; i < forks.length; i++) {
            forks[i] = new Semaphore(1);
        }
    }

    public void takeFork(int forkIndex) {
        try {
            forks[forkIndex].acquire();
        } catch (InterruptedException e) {
            throw new RuntimeException(e);
        }
    }

    public void releaseFork(int forkIndex) {
        forks[forkIndex].release();
    }
}